lib/p2p/NodeList.ts

```diff
@@ -137,10 +139,12 @@ class NodeList extends EventEmitter {
 
       if (node.reputationScore < NodeList.banThreshold) {
         promises.push(this.setBanStatus(node, true));
+        this.bannedNodes.add(node.nodePubKey);
```

Comment:
We want to delete the banned node from `this.nodes` here. This is to ensure we don't share the banned node with other peers and to be consistent with the loading logic which only adds unbanned nodes to `this.nodes`.